cve_bin_tool/checkers/perl.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python3
+
+"""
+CVE checker for Perl
+References:
+https://www.cvedetails.com/vulnerability-list/vendor_id-1885/Perl.html
+"""
+from ..util import regex_find
+import re
+
+
+def guess_version(lines):
+    regex1 = [r"perl-[45]:([5]+\.[0-9]+\.[0-9]+)", r"perl-[45]:([5]+\.[0-9]+)"]
+    match = regex_find(lines, *regex1)
+    if match != "UNKNOWN":
+        return match
+    for line in lines:
+        match = re.sub(
```

Comment:
Take a look again at how regex_find does this -- you're overwriting your match variable every time, so I think this might only work if the match happened on the very last line of the list.  You'll need a temproary variable and compare until you find something, or to just return the first match found so it breaks out of the function before anything is overwritten.